### ISSUE TITLE
False positive `possibly-used-before-assignment` for `-> typing.NoReturn`

### ISSUE THREAD

**Issue by kdelay** [PyCQA/pylint#11271]
`-> NoReturn` is treated as terminating (#9674 / #9714), but the qualified form
`-> typing.NoReturn` is not. `typing.Never` behaves the same.

```python
import typing

def terminate(msg) -> typing.NoReturn:
    raise SystemExit(msg)

def f(flag):
    if flag:
        cmd = "ls"
    else:
        terminate("nope")
    return cmd  # E0606
```

`utils.is_terminating_func` requires `isinstance(inferred.returns, nodes.Name)`, so the
`nodes.Attribute` annotation never reaches the qname check that already knows
`typing.NoReturn`. `RefactoringChecker._is_function_def_never_returning` accepts both,
so `inconsistent-return-statements` is unaffected.

### Expected behavior

No message, and `unreachable` after such a call (same cause).

### Pylint version

pylint 4.0.7, astroid 4.0.4, Python 3.14.3; also on main 03eaf44a.
